### PR TITLE
sort years in descending order, defaulting to most recent year

### PR DESCRIPTION
**Pull request by cfhull**
https://github.com/Hyperobjekt/seeing-solitary/issues/133

### CODE REVIEW

File: normalizeData.js [hunk 1]
```diff
@@ -360,13 +360,16 @@ async function getTransTotals(last) {
...
+}
+
 async function getPregnantTotals(last) {
-  last['summary']["pregnant"] = {
-    2015: [],
-    2021: [],
-  }
   last["jurisdictions"].forEach((j) => {
-    ["2015", "2021"].forEach((year) => {
+    ;["2015", "2021"].forEach((year) => {
```
Comment (cfhull) [inline normalizeData.js:372]: prettier autoformat

File: normalizeData.js [hunk 2]
```diff
@@ -395,7 +398,7 @@ async function getPregnantTotals(last) {
           ),
         }
       }
-      last['summary']["pregnant"][year].push(stub)
+      last["summary"]["pregnant"][year].push(stub)
```
Comment (cfhull) [inline normalizeData.js:401]: prettier autoformat

File: normalizeData.js [hunk 3]
```diff
@@ -360,13 +360,20 @@ async function getTransTotals(last) {
   return summary
 }
 
+async function descSortYears(last) {
```
Comment (cfhull) [inline normalizeData.js:363]: I manually called this function from the normalize data function, and ran this file with node to generate the data json file.

It simply sorts the years descending so the dropdowns go from most recent to least recent, with the most recent as default.

**Comment by lizlove** [comment]
So I think the issue here is not the internal data sorting, but the appearance of the dropdowns? Data has also been updated subsequent to this several times so I'm going to hold off on merging this one until we get a bit more clarity. Might be more useful to update the select years dropdown component.

**Comment by cfhull** [comment]
> So I think the issue here is not the internal data sorting, but the appearance of the dropdowns? Data has also been updated subsequent to this several times so I'm going to hold off on merging this one until we get a bit more clarity. Might be more useful to update the select years dropdown component.

I would have just sorted the years in the `SelectYear` component, but the `activeYear` gets passed in from many places, often multiple levels up. I thought it would be easier to just sort the data during the normalization process, rather than deal with it in multiple places.

I can either:
1. Pull the latest data, and re-run the sort years function to fix the merge conflicts.
2. Sort the years in the `SelectYear` component, and replace all instances of activeYear to use the highest value instead of the first value

Let me know which you'd prefer.

**Comment by cfhull** [comment]
@lizlove Awaiting your thoughts on this. Thanks!

**Review by lizlove — APPROVED** [review]

**Pull request merged**